sound/soc/sof/intel/hda-pcm.c

```diff
@@ -235,5 +249,15 @@ int hda_dsp_pcm_close(struct snd_sof_dev *sdev,
 
 	/* unbinding pcm substream to hda stream */
 	substream->runtime->private_data = NULL;
+
+	spcm = snd_sof_find_spcm_dai(sdev, rtd);
+	if (!spcm)
+		return -EINVAL;
+
+	if (!spcm->stream[substream->stream].d0i3_compatible) {
+		pm_runtime_mark_last_busy(&hdev->d0ix_dev->dev);
+		pm_runtime_put_autosuspend(&hdev->d0ix_dev->dev);
```

Comment:
I've stated this before, but we should not mix layers. Each device should take care of its own power management, you are abusing the pm_runtime framework here.
I am not ok with this.